include/natalie/gc/allocator.hpp

```diff
@@ -104,15 +101,15 @@ class Allocator {
     HeapBlock *add_heap_block() {
         auto *block = reinterpret_cast<HeapBlock *>(aligned_alloc(HEAP_BLOCK_SIZE, HEAP_BLOCK_SIZE));
         new (block) HeapBlock(m_cell_size);
-        m_blocks.push(block);
+        m_blocks.put(block, block);
```

Comment:
You can treat the hashmap as a hashset by using `set()` instead of `put()`.

Reply:
Ahh that makes a lot of sense! Will fix that